modules/ai/assets/js/editor/hooks/use-upgrade-message.js

```diff
@@ -0,0 +1,17 @@
+import useIntroduction from './use-introduction';
+
+export const USAGE_PERCENTAGE_THRESHOLD = 80;
+
+const useUpgradeMessage = ( { usagePercentage, hasSubscription } ) => {
+	const { isViewed, markAsViewed } = useIntroduction( 'e-ai-upgrade-message' );
+	const showBadge = usagePercentage >= USAGE_PERCENTAGE_THRESHOLD || ( ! hasSubscription && isViewed );
+	const showBanner = usagePercentage < USAGE_PERCENTAGE_THRESHOLD && ! hasSubscription && ! isViewed;
```

Comment:
maybe separating it into regular conditions will make it more readable?

```JS
const isAboveThershold = usagePercentage >= USAGE_PERCENTAGE_THRESHOLD;
const isBelowTherhold = usagePercentage < USAGE_PERCENTAGE_THRESHOLD;

let showBadge = false;
let showBanner = false;

if ( hasSubscription ) {
	showBadge = isAboveThershold;
	showBanner = false;
} else {
	showBadge = isBannerViewd;
	showBanner = isBelowTherhold;
}

if ( showBadge ) {
	showBanner = false;
}
```

Reply:
Done.